private/pkg/storage/copy.go

```diff
@@ -94,11 +96,22 @@ func CopyWithExternalPaths() CopyOption {
 	}
 }
 
+// CopyWithWriteToExternalPaths returns a new CopyOption that says to copy and write
+// to external paths.
+//
+// The to WriteBucket must support setting external paths.
+func CopyWithWriteToExternalPaths() CopyOption {
+	return func(copyOptions *copyOptions) {
+		copyOptions.writeToExternalPaths = true
+	}
+}
```

Comment:
Flagging this one for discussion @bufdev - it's currently used to write the files in-place based on the files' external paths with `storage.Copy` (see `private/buf/cmd/buf/command/format/format.go`).

Reply:
I don't quite understand it at first glance, can talk offline